lib/kb_Metrics/metricsdb_controller.py

```diff
@@ -112,8 +112,7 @@ def update_user_info(self, requesting_user, params, token):
         idKeys = ['username', 'email']
         dataKeys = ['full_name', 'signup_at', 'last_signin_at', 'roles']
         for u_data in auth2_ret:
-            filterByKey = lambda keys: {x: u_data[x] for x in keys}
-            idData = filterByKey(idKeys)
+            (idData, userData) = _partition_by_keys(u_data, idKeys, dataKeys)
```

Comment:
I don't know how did you successfully test this while _partition_by_keys has obvious bugs.